src/wp-includes/rest-api/endpoints/class-wp-rest-block-renderer-controller.php

```diff
@@ -34,47 +34,52 @@ public function __construct() {
 	 * @see register_rest_route()
 	 */
 	public function register_routes() {
-		$block_types = WP_Block_Type_Registry::get_instance()->get_all_registered();
-
-		foreach ( $block_types as $block_type ) {
-			if ( ! $block_type->is_dynamic() ) {
-				continue;
-			}
-
-			register_rest_route(
-				$this->namespace,
-				'/' . $this->rest_base . '/(?P<name>' . $block_type->name . ')',
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>[a-z0-9-]+/[a-z0-9-]+)',
+			array(
+				'args'   => array(
+					'name' => array(
+						'description' => __( 'Unique registered name for the block.' ),
+						'type'        => 'string',
+					),
+				),
 				array(
-					'args'   => array(
-						'name' => array(
-							'description' => __( 'Unique registered name for the block.' ),
-							'type'        => 'string',
+					'methods'             => array( WP_REST_Server::READABLE, WP_REST_Server::CREATABLE ),
+					'callback'            => array( $this, 'get_item' ),
+					'permission_callback' => array( $this, 'get_item_permissions_check' ),
+					'args'                => array(
+						'context'    => $this->get_context_param( array( 'default' => 'view' ) ),
+						'attributes' => array(
+							'description'       => __( 'Attributes for the block' ),
+							'type'              => 'object',
+							'default'           => array(),
+							'validate_callback' => static function ( $value, $request ) {
+								$block = WP_Block_Type_Registry::get_instance()->get_registered( $request['name'] );
+
+								if ( ! $block ) {
+									// This will get rejected in ::get_item().
```

Comment:
Do you think that `$block_type->is_dynamic()` is no longer necessary? I would consider keeping it to make sure it only works with blocks that define `render_callback`. Unless it returns the content of the block otherwise which might be useful in the context outside of WordPress.

Reply:
No I think you're right, it makes more sense to keep that check. It'll just end up returning an empty string otherwise, AFAICT, since we don't support sending inner content of any kind. I'll add it in the `get_item` callback.

Reply:
Fixed in 45e8026.